src/qat/purr/backends/qiskit_simulator.py

```diff
@@ -267,7 +271,30 @@ def execute(self, builder: QiskitBuilder) -> Dict[str, Any]:
                 returns.extend(inst.variables)
         if len(assigns) > 1:
             log.warning("Qasm Simulator does not support multiple assignment.")
-        return {key: trimmed for key in returns} if returns else trimmed
+        if returns:
+            # trim the qiskit returns to the classical register
+            removals = self.model.qubit_count - len(assigns[returns[0]])
```

Comment:
maybe assign `assigns[returns[0]]` to a variable to give it a friendly name for readability?

Reply:
(as you use it a couple of times)